v2/internal/controllers/samples_test.go

```diff
@@ -0,0 +1,230 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/emirpasic/gods/maps/linkedhashmap"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/Azure/azure-service-operator/v2/internal/reflecthelpers"
+	"github.com/Azure/azure-service-operator/v2/internal/resolver"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+)
+
+var cases = []struct {
+	Group          string
+	testName       string
+	UseRandomName  bool
+	DeleteChildren bool
+}{
```

Comment:
Still need to move this to a directory-scan based approach?

Reply:
With the directory scan approach we can't set the settings flags per test group as we can right now.

Reply:
Is there a reason that we can't just set `UseRandomName` to true for everybody?

Reply:
For the `DeleteChildren` flag, I think we could maybe get rid of it entirely and just always delete the RG when we're done, rather than exercising delete on each thing individually?

The intent of these tests is to test the samples themselves more than the reconciler behavior. We already cover this in the handcrafted tests so for the samples a simpler deletion approach seems ok since we're more interested in the `kubectl apply` anyway?

That (I think) removes both of these flags and enables you to do a directory scan instead?

Reply:
For a few with references present, we can't set random name for all of them